### ISSUE TITLE
Bring over unmerged PRs from ghcide

### ISSUE THREAD

**Issue by wz1000** [haskell/haskell-language-server#705]
This issue exists to track progress on the following PRs that were open on ghcide but not merged:

- [X] https://github.com/haskell/ghcide/pull/857 (#1284)
- [X] https://github.com/haskell/ghcide/pull/861
- [X] https://github.com/haskell/ghcide/pull/888
- [X] https://github.com/haskell/ghcide/pull/889: tracked in its own issue #532
- [X] https://github.com/haskell/ghcide/pull/898 (#704)
- [X] ~~https://github.com/haskell/ghcide/pull/891~~ (not correct)
- [X] https://github.com/haskell/ghcide/pull/897 (#1235)

**Comment by jneira** [comment]
Are all of them pending?
//cc @bubba @georgefst @expipiplus1 @DunetsNM

**Comment by expipiplus1** [comment]
Happy to ignore 888 for now, it can be worked around with GHC flags anyway.

**Comment by georgefst** [comment]
Yes, I actually got half way to reopening haskell/ghcide#897 last night - just need to fix up a test on older GHCs.

**Comment by DunetsNM** [comment]
let's ignore the haskell/ghcide#861 for now, as I'm blocked by upstream haddock anyway for indefinite time. By the time when it's fixed (if ever) hls sources may drift a lot.

btw, if I can fix and publish the haddock upstream myself I'd be happy to do so, but it seems to lack owners, at least I had no response on the issues reported

**Comment by jneira** [comment]
Afaics only https://github.com/haskell/ghcide/pull/889 would need reopened here
//cc @wz1000

**Comment by jneira** [comment]
I think we could close thisone, only left https://github.com/haskell/ghcide/pull/889 and we can track it in its own issue
